test/controllers/supplementary/AdditionalInformationControllerSpec.scala

```diff
@@ -171,27 +181,42 @@ class AdditionalInformationControllerSpec
 
           val body = Seq(("code", createRandomString(6)), ("description", ""), addActionURLEncoded)
           val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).get
+          val page = contentAsString(result)
 
           status(result) must be(BAD_REQUEST)
-          contentAsString(result) must include(messages(codeError))
+
+          checkErrorsSummary(page)
+          checkErrorLink(page, 1, codeError, "#code")
+
+          getElementByCss(page, "#error-message-code-input").text() must be(messages(codeError))
```

Comment:
:+1: I like this approach. 
We need to move into this approach, by correctly identifying the HTML elements in a page.